src/libraries/apiclient/apiclientcore.js

```diff
@@ -1426,25 +1438,74 @@ define(["events", "appStorage"], function(events, appStorage) {
         })
     }, ApiClient.prototype.reportPlaybackProgress = function(options) {
         if (!options) throw new Error("null options");
-        var newPositionTicks = options.PositionTicks;
-        if ("timeupdate" === (options.EventName || "timeupdate")) {
-            var now = (new Date).getTime(),
-                msSinceLastReport = now - (this.lastPlaybackProgressReport || 0);
-            if (msSinceLastReport <= 1e4) {
-                if (!newPositionTicks) return Promise.resolve();
-                var expectedReportTicks = 1e4 * msSinceLastReport + (this.lastPlaybackProgressReportTicks || 0);
-                if (Math.abs((newPositionTicks || 0) - expectedReportTicks) < 5e7) return Promise.resolve()
-            }
-            this.lastPlaybackProgressReport = now
-        } else this.lastPlaybackProgressReport = 0;
-        this.lastPlaybackProgressReportTicks = newPositionTicks;
-        var url = this.getUrl("Sessions/Playing/Progress");
-        return this.ajax({
-            type: "POST",
-            data: JSON.stringify(options),
-            contentType: "application/json",
-            url: url
-        })
+
+        const eventName = options.EventName || "timeupdate";
+        let reportRateLimitTime = reportRateLimits[eventName] || 0;
+
+        const now = (new Date).getTime();
+        const msSinceLastReport = now - (this.lastPlaybackProgressReport || 0);
+        const newPositionTicks = options.PositionTicks;
+
+        if (msSinceLastReport < reportRateLimitTime && eventName === "timeupdate" && newPositionTicks) {
+            const expectedReportTicks = 1e4 * msSinceLastReport + (this.lastPlaybackProgressReportTicks || 0);
```

Comment:
```suggestion
            const expectedReportTicks = reportRateLimitTime * msSinceLastReport + (this.lastPlaybackProgressReportTicks || 0);
```
you've extracted this `1e4` into global constant already ;)

Reply:
No, this is `ms` to `tick` conversion. Here approximately calculating how much ticks have been played for elapsed time.